src/main/webapp/js/components/fields/field-config-modal.jsx

```diff
@@ -0,0 +1,286 @@
+        />
+      </span>
+    );
+  }
+
+  drawSelect(displayName, name, val, options) {
+    return (
+      <span className="input-row">
+        <span className="field-config-label">{displayName}</span>
+        <Select
+          onChange={value => this.handleChange(value, name)}
+          options={FIELD_TYPE_OPTIONS}
+          defaultValue={getDefaultSelectOption(options, val)}
+        />
+      </span>
+    );
+  }
+
+  drawInput(displayName, name, val) {
+    return (
+      <span className="input-row">
+        <span className="field-config-label">{displayName}</span>
+        <Input
+          onChange={value => this.handleChange(value, name)}
+          value={val || ''}
+        />
+      </span>
+    );
+  }
+
+  displayFields(item) {
+    const {
+      base, editable, filterable, required, fieldType, entity,
+      displayName, name, format, relatedEntity, relatedField,
+    } = this.state;
+    return (
+      !!item && (
+      <div>
+        <Form
+          onSubmit={this.onSubmit}
+          render={({ handleSubmit }) => (
+            <form onSubmit={handleSubmit} className="modal-fields">
+              { this.drawCheckbox('Base', 'base', base) }
+              { this.drawCheckbox('Editable', 'editable', editable) }
+              { this.drawCheckbox('Filterable', 'filterable', filterable) }
+              { this.drawCheckbox('Required', 'required', required) }
+              { this.drawSelect('Field type', 'fieldType', fieldType, FIELD_TYPE_OPTIONS) }
+              { this.drawSelect('Entity', 'entity', entity, ENTITY_TYPE_OPTION) }
+              { this.drawSelect('Related entity', 'relatedEntity', relatedEntity, ENTITY_TYPE_OPTION) }
+              { this.drawInput('Display name', 'displayName', displayName) }
+              { this.drawInput('Name', 'name', name) }
+              { this.drawInput('Format', 'format', format) }
+              { this.drawInput('Related field', 'relatedField', relatedField) }
+              <div>
+                <span>
+                  <button type="submit" className="btn btn-primary">Save</button>
+                </span>
+                <span style={{ float: 'right' }}>
+                  <button type="button" className="btn btn-danger" onClick={(event) => { event.preventDefault(); window.confirm('Are you sure you wish to delete this item?'); }}>Delete</button>
+                </span>
+              </div>
+            </form>
+          )}
+        />
+      </div>
+      )
+    );
+  }
+
+  mapConfigToState(item) {
+    if (item) {
+      const {
+        base, editable, filterable, required, fieldType, entity,
+        displayName, name, relatedEntity, relatedField, format,
+      } = item;
+      this.setState({
+        base,
+        editable,
+        filterable,
+        required,
+        fieldType,
+        entity,
+        displayName,
+        name,
+        relatedEntity,
+        relatedField,
+        format,
+      });
+    }
+  }
+
+  handleChange(val, name) {
```

Comment:
remove this method, you should not change the state like this.
form will take care of state changes (it has its own state for this)